cli/deployment/config.go

```diff
@@ -0,0 +1,127 @@
+package deployment
+
+import (
+	"fmt"
+	"log"
+	"regexp"
+	"strings"
+
+	"gopkg.in/yaml.v2"
+)
+
+// patern to parse $(project.deployment.resource.name) and $(deployment.resource.name)
+var pattern = regexp.MustCompile(`\$\(out\.(?P<token>[-.a-zA-Z0-9]+)\)`)
+
+// Config struct keep config data parsed from passed config.yaml
+type Config struct {
+	Name        string
+	Project     string
+	Description string
+	Imports     []struct {
+		Name string
+		Path string
+	}
+	Resources []interface{}
+	file      string
+	data      string
+}
+
+// NewConfig creates new Config object from provided yaml file
+func NewConfig(data string, file string) Config {
+	config := Config{
+		file: file,
+		data: data,
+	}
+
+	err := yaml.Unmarshal([]byte(data), &config)
+	if err != nil {
+		log.Fatalf("error: %v", err)
+	}
+	return config
+}
+
+// FullName returns name in form of ProjectName.DeploymentName, this name should be unique and it could be used as map key
+// for maps like map[string]Config
+func (c Config) FullName() string {
+	return c.Project + "." + c.Name
+}
+
+func (c Config) String() string {
+	return c.FullName()
+}
+
+// YAML function converts Config object to yaml
+// overrides all relative paths for imports to absolute form,
+// removes all custom elements gcloud deployment manager not aware of (name, project, description)
```

Comment:
"The YAML function marshals a Config object to YAML. It sets all relative import paths to absolute paths and removes all custom elements that the gcloud deployment manager is not aware of, including name, project, and description."